e2eTests/test_task_journey.py

```diff
@@ -0,0 +1,93 @@
+from time import sleep
+from threading import Thread
+
+import vcr
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions
+
+import importlib
+import os
+import pytest
+
+from e2eTests.utils.temp_trello_boards import create_trello_board, delete_trello_board
+from todo_app import app
+from dotenv import find_dotenv, load_dotenv
+import todo_app.data.trello_client
+
+
+@pytest.fixture(scope='module')
+def app_with_temp_board():
+    # Create the new board & update the board id environment variable
+    with vcr.use_cassette(
```

Comment:
Interesting choice to use VCR with the end-to-end test; typically I'd expect it to line up with the integration test in place of the explicit mocking, but I quite like it here: although it's no longer a full end-to-end, it still gives us the value of checking that the web pages are manipulated successfully, whilst giving us the ability to check that the API calls all line up with reality whenever we re-create the cassette.

Reply:
Ah, I must have misunderstood that bit of the stretch goals. In my head this was a nice way of allowing the end-to-end tests to run in a pipeline without that pipeline having to call an external API, whereas for the integration tests I thought we'd still want that more fine-grained control of the mocked API responses from the stubs.